lib/constraint_solver/solver.ml

```diff
@@ -46,6 +58,37 @@ let rec gtype_of_type : state:State.t -> env:Env.t -> C.Type.t -> G.Type.t =
       (Constr (List.map args ~f:self, constr))
 ;;
 
+let match_type
+  :  state:State.t -> env:Env.t -> G.Type.t Structure.Former.t
+  -> C.Type.Scruintee.t * Env.t
```

Comment:
Suggest `Env.t * C.Type.Matchee.t` as the return type (less de-structuring required)